### ISSUE TITLE
rematch/select - store.select types incorrect

### ISSUE THREAD

**Issue by andrejpavlovic** [rematch/rematch#866]
**Describe the bug**
Adding selectors to a model and then trying to use them, gives unexpected typing.

For example, assuming we have `myModel` with a `total` selector, one would expect that `selection` type of the code below to be `(state: any) => {total: number}`, but instead it is actually `(state: any) => {total: (state: any) => number}`.
```typescript
const selection = store.select((models) => ({
  total: models.myModel.totalIds
}));
```

**Reproduce Link**
https://codesandbox.io/s/rematch-select-wrong-typing-5tw4w?file=/src/index.ts

**To Reproduce**
1. Go to reproduce link
2. Scroll to bottom of `index.ts` file
3. See the typescript error.

**Expected behavior**
Selector value should not be a function that expects model state that returns the value.

**Screenshots**
![2021-02-18_231008](https://user-images.githubusercontent.com/983644/108456521-80d65100-723e-11eb-920c-7878c6e5ad58.png)

**Desktop**

- OS: Win10
- Browser: edge
- Version: 88

**Comment by semoal** [comment]
Hi Andrej, yes actually the select types probably has some pending work to fit all the cases. 
I'm glad that you found this issue and you're helping us debuggin this.

If you want to take a look and submit an awesome pull request to fix this, here is the select plugin types :) https://github.com/rematch/rematch/blob/main/packages/select/src/types.ts

**Comment by masiya255** [comment]
@andrejpavlovic https://github.com/rematch/rematch/issues/865 here I described a similar problem, we are waiting for the fix!

**Comment by semoal** [comment]
@andrejpavlovic  do you find any situation where we need to adjust rematch v2 to complete your migration? 
I'll planed to release tomorrow all the changes we made :)

**Comment by andrejpavlovic** [comment]
I was able to complete most of the migration at this point. Will let you know if anything else comes up :) Thanks for staying on top of this!

**Comment by semoal** [comment]
Released 2.0.1 and 3.0.1 for @rematch/select guys! I hope everything runs smoothly for everyone.

Feel free to create or reopen anything you find.
